chemprop/utils/mixins.py

```diff
@@ -1,20 +1,57 @@
+"""Mixin classes for the project."""
+
 import inspect
 from typing import Any, Collection
 
 
 class ReprMixin:
     def __repr__(self) -> str:
-        items = self.get_params()
+        """Return a string representation of the object.
+
+        Returns
+        -------
+        str
+            The string representation of the object.
+        """
+        param_repr_list = []
+        for param_name, param_value in self.get_params(deep=False).items():
+            param_default = self._get_default_params()[param_name]
+            if param_value != param_default:
+                param_repr_list.append(f"{param_name}={repr(param_value)}")
+
+        non_default_params = ", ".join(param_repr_list)
+        return f"{self.__class__.__name__}({non_default_params})"
+
+    def _get_default_params(self) -> dict[str, Any]:
+        """Get the default parameters for initializing the class."""
+        sig = inspect.signature(self.__class__)
+        return {k: v.default for k, v in sig.parameters.items()}
+
+    def _get_parm_names(self) -> list[str]:
+        """Get the default parameters for initializing the class."""
+        return sorted(self._get_default_params().keys())
```

Comment:
why `sorted`?

Reply:
scikit-learn does sorted, maybe to be easier to understand if there are lots of keys? I don't have a preference either way.

Reply:
IMO it's clearer without `sorted`--then the list of param names follows the initializer signature, i.e., the source.

Reply:
Copied from the [answer above](https://github.com/chemprop/chemprop/pull/649#discussion_r1492516015):  
I used sorted, as the `keys()` method returns a `dict_keys` object and not a `list`.
(Using list would equally work, but I had some years ago problems with reproducing results exactly, where scripts used sets. So maybe I am just a bit paranoid.)